Source/JavaScriptCore/dfg/DFGConstantFoldingPhase.cpp

```diff
@@ -958,12 +958,7 @@ class ConstantFoldingPhase : public Phase {
                 if (radixValue && radixValue.isInt32()) {
                     int32_t radix = radixValue.asInt32();
                     if (2 <= radix && radix <= 36) {
-                        if (radix == 10) {
-                            node->setOpAndDefaultFlags(ToString);
-                            node->clearFlags(NodeMustGenerate);
-                            node->child2() = Edge();
-                        } else
-                            node->convertToNumberToStringWithValidRadixConstant(radix);
+                        node->convertToNumberToStringWithValidRadixConstant(radix);
```

Comment:
Also, let's attempt to make it `ToString` if the speculation of the `child1` is some number types (`Int32Use`, `Int52Use` etc.).